uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/BatchIndex.java

```diff
@@ -91,151 +92,115 @@ public void run() {
-                                                 if (indexer.supportsPath(file)) {
-                                                     final KObject kObject = indexer.toKObject(file);
-                                                     if (kObject != null) {
-                                                         if (!indexDisposed.get()) {
-                                                             indexEngine.index(kObject);
-                                                         } else {
-                                                             return FileVisitResult.TERMINATE;
-                                                         }
-                                                     }
-                                                 }
-                                             }
-                                         }
-                                     }
-                                 } catch (final Exception ex) {
-                                     if (indexDisposed.get()) {
-                                         logWarning("Batch index couldn't finish. [@" + root.toUri().toString() + "]");
-                                         return FileVisitResult.TERMINATE;
-                                     } else {
-                                         logError("Index fails. [@" + file.toString() + "]",
-                                                  ex);
-                                     }
-                                 }
-                                 if (indexDisposed.get()) {
-                                     return FileVisitResult.TERMINATE;
-                                 }
-                                 return FileVisitResult.CONTINUE;
-                             }
-                         });
+            }
 
             if (!indexDisposed.get()) {
-                logInformation("Completed indexing of " + root.toUri());
-                if (callback != null) {
-                    callback.run();
-                }
-            } else {
-                logWarning("Batch index couldn't finish. [@" + root.toUri().toString() + "]");
-            }
-        } catch (final IllegalStateException ex) {
-            if (indexDisposed.get()) {
-                logWarning("Batch index couldn't finish. [@" + root.toUri().toString() + "]");
+                logInformation("Starting indexing of " + fs.getName() + " ...");
+                dispatcher.schedule(executorService)
+                .thenRun(() -> {
+                    logInformation("Completed indexing of " + fs.getName());
+                    if (callback != null) {
+                        callback.run();
+                    }
+                })
+                .exceptionally(ex -> {
+                    try {
+                        throw ex;
+                    } catch (DisposedException de) {
+                        logWarning("Batch index couldn't finish. [@" + fs.getName() + "]");
```

Comment:
Does it make sense to add exception to log message here as well as it's done in following catch branches? Similar applies to other `logWarning` calls here..

Reply:
@jhrcek All of the warning messages are for situations where indexing is terminated before completion. This was the same behaviour that existed prior to the change-set.

Maybe these should be errors, but I don't think they have anything to do with the subject matter of this PR so I'd prefer to open a separate issue for this.